From 580ef0dff37f063e1a4898528ce3a5347789bbc1 Mon Sep 17 00:00:00 2001
From: Ramanathan Choodamani <quic_rchoodam@quicinc.com>
Date: Tue, 7 Nov 2023 15:49:18 -0800
Subject: [PATCH] wifi: ath12k: PPEDS - Add ENCAP flags during tx in slow path

When packets are transmitted through the link ndev slow path,
we do not add ENCAP flags currently to the msdu, which in
turn wrongly categorize the packet completions.

Add hw encap flags to the msdu when packets are routed in
the default slow path specific to the link_ndev

Signed-off-by: Ramanathan Choodamani <quic_rchoodam@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/bondif.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/wireless/ath/ath12k/bondif.c
+++ b/drivers/net/wireless/ath/ath12k/bondif.c
@@ -891,6 +891,7 @@ netdev_tx_t ieee80211_link_subif_start_x
 	int is_mcast, ring_id = 0;
 	struct ethhdr *eh = NULL;
 	struct ieee80211_tx_info *info = IEEE80211_SKB_CB(skb);
+	struct ath12k_skb_cb *skb_cb = ATH12K_SKB_CB(skb);
 
 	ieee80211_tx_stats_bonded_if(dev, skb->len);
 
@@ -901,7 +902,9 @@ netdev_tx_t ieee80211_link_subif_start_x
 
 	eh = (struct ethhdr *)skb->data;
 	is_mcast = is_multicast_ether_addr(eh->h_dest);
-	ATH12K_SKB_CB(skb)->link_id = arvif->link_id;
+	skb_cb->link_id = arvif->link_id;
+	skb_cb->flags |= ATH12K_SKB_HW_80211_ENCAP;
+	info->flags |= IEEE80211_TX_CTL_HW_80211_ENCAP;
 
 	if (!arvif_pvt->is_started) {
 		ath12k_dbg(NULL, ATH12K_DBG_PPE, "Dropping frames as arvif is not started\n");
